index.js

```diff
@@ -24,12 +24,14 @@ app.get('/:sheet', (req, res) => {
   let doc = new GoogleSpreadsheet(sheet_id);
   let sheet_data = [];
   doc.useServiceAccountAuth(creds, err => {
-    if (err) res.send(err);
-    doc.getRows(1, (err, rows) => {
-      if (rows[0] == undefined) {
-        return res.send('');
-      }
-      if (err) res.send(err);
+    if (err) throw new Error('Unable to connect to docs');
```

Comment:
These errors should be returned to the frontend IMO with the exact message as you wrote for throwing error. Setting the status code as 500 would make the catch block active on the frontend and we will have the error console logged as well as have a response in the frontend.

Reply:
@thealphadollar  catch block becomes active only when there is a netwrok error, and 4xx and 5xx aren't network errors. See [the 1st answer](https://stackoverflow.com/questions/38235715/fetch-reject-promise-and-catch-the-error-if-status-is-not-ok) and [this](https://support.loader.io/article/30-error-network-faq). What we can do it send a error message like `res.send('Unable to connect  to docs')` and show that only in frontend `then` block throw and error which will redirect to their catch block or directly show to the errror in `then` block only